File: mobile/src/bindings/grocery.ts
import { invoke } from "@tauri-apps/api/core";

export const grocery = {
  async plan(name: string): Promise<string> {
    return await invoke("grocery", { action: "Plan", name });
  },

  async meal(name: string): Promise<string> {
    return await invoke("grocery", { action: "Meal", name });
  },
};

File: mobile/src-tauri/src/commands/grocery.rs
use foody_core::app::App;
use serde::Deserialize;
use tauri::State;

use crate::commands::serialize_json;

#[derive(Deserialize)]
#[serde(tag = "action")]
pub enum GroceryCommand {
    Plan { name: String },
    Meal { name: String },
}

#[tauri::command]
pub async fn grocery(
    state: State<'_, App>,
    command: GroceryCommand,
) -> Result<serde_json::Value, String> {
    match command {
        GroceryCommand::Plan { name } => serialize_json(state.grocery_plan(&name)).await,
        GroceryCommand::Meal { name } => serialize_json(state.grocery_meal(&name)).await,
    }
}
